packages/smithy/smithy/lib/src/http/http_operation.dart

```diff
@@ -268,10 +268,11 @@ abstract class HttpOperation<InputPayload, Input, OutputPayload, Output>
     var successCode = this.successCode();
     try {
       final payload = await protocol.deserialize(response.split());
+      // ignore: unnecessary_cast
       output = switch (payload) {
         Output _ => payload,
         _ => buildOutput(payload, response),
-      };
+      } as Output;
```

Comment:
I think this is the preferred fix.

```dart
      output = switch (payload) {
        final Output p => p,
        _ => buildOutput(payload, response),
      };
```

The issue with the statement currently is that line in line 273 `payload` is type `OutputPayload`. We know it is type `Output` (line 273 would not execute if it were not) so the cast here should be safe, but I think we can avoid it by returning `p` which the dart compilers knows is type `Output`.